packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ParameterProvider.cs

```diff
@@ -91,39 +88,35 @@ private ParameterValidationType GetParameterValidation(InputModelProperty proper
             return ParameterValidationType.AssertNotNull;
         }
 
-        private struct ParameterByTypeEqualityComparer : IEqualityComparer<ParameterProvider>
+        public override bool Equals(object? obj)
         {
-            public bool Equals(ParameterProvider? x, ParameterProvider? y)
-            {
-                return Equals(x?.Type, y?.Type);
-            }
-
-            public int GetHashCode([DisallowNull] ParameterProvider obj) => obj.Type.GetHashCode();
+            return obj is ParameterProvider parameter && Equals(parameter);
         }
 
-        private class ParameterTypeAndNameEqualityComparer : IEqualityComparer<ParameterProvider>
+        public bool Equals(ParameterProvider? y)
```

Comment:
Can we add some unit tests around this equality so we know we are correct in all cases?